### PR TITLE
docs: sptagindexer crud

### PR DESCRIPTION
**Pull request by florian-hoenicke**

### CODE REVIEW

File: indexers/vector/SptagIndexer/__init__.py [hunk 1]
```diff
@@ -94,6 +98,10 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
             return _index
 
     def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.ndarray', 'np.ndarray']:
+        """Find the top-k vectors with smallest ``metric`` and return their ids in ascending order.
```
Comment (JoanFM) [inline indexers/vector/SptagIndexer/__init__.py:101]: have u checked if this applies to every metric or if there is an exception? for instance with FaissIndexer the inner product returns similarity, (maybe we should transform to distance ourselves)

**Review by JoanFM — COMMENTED** [review]
are u sure you are aligned with latest master? https://github.com/jina-ai/jina-hub/pull/3501/files

**Review by JoanFM — COMMENTED** [review]
ah ok I see now

File: indexers/vector/SptagIndexer/Dockerfile [hunk 2]
```diff
@@ -12,10 +12,8 @@ RUN wget https://cmake.org/files/v3.15/cmake-3.15.5.tar.gz && \
     make -j2 && \
     make install
 
-RUN git clone https://github.com/microsoft/SPTAG.git
+RUN git clone https://github.com/microsoft/SPTAG.git/tree/a91ed49a041a24bd145874d3b3c815bde0430e65
```
Comment (JoanFM) [inline indexers/vector/SptagIndexer/Dockerfile:15]: can we merge these 2 RUN layers?

Comment (florian-hoenicke) [reply indexers/vector/SptagIndexer/Dockerfile:15]: I thought so but have to reverse the commit

File: indexers/vector/SptagIndexer/__init__.py (hunk 1 above)
Comment (florian-hoenicke) [reply indexers/vector/SptagIndexer/__init__.py:103]: Good point.
I checked the documentation:
`Vectors returned for a query vector are the vectors that have smallest L2 distance or cosine distances with the query vector.`
Only cosine and l2 are supported.
Also, I added an assertion to the test to make sure the order is ascending.

**Review by theUnkownName — APPROVED** [review]

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**